ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/NormalForm/StrictMVar.hs

```diff
@@ -85,7 +92,11 @@ newEmptyMVarWithInvariant inv =
 -------------------------------------------------------------------------------}
 
 noThunksInvariant :: NoThunks a => a -> Maybe String
+#ifdef CHECK_NOTHUNKS_INVARIANT
 noThunksInvariant = fmap show . unsafeNoThunks
+#else
+noThunksInvariant = const Nothing
+#endif
```

Comment:
>  It might be a good idea to extract noThunksInvariant into its own module Ouroboros.Consensus.Util.NormalForm.Invariant. This would allows us to reduce the amount of CPP needed to conditionally take the different build paths without raising warnings.

Agreed, that sounds like a good idea :+1: